rita_common/src/usage_tracker/mod.rs

```diff
@@ -116,8 +117,47 @@ pub struct UsageTracker {
     payments: VecDeque<PaymentHour>,
 }
 
+fn check_usage_hour(input: &mut VecDeque<UsageHour>, last_save_hour: u64) -> bool {
+    let number_of_bytes = 10000;
+    let mut total_used_bytes = 0;
+    while !input.is_empty() {
+        match input.pop_front() {
+            Some(val) => {
+                total_used_bytes += val.up;
+                if val.index < last_save_hour {
+                    return total_used_bytes < number_of_bytes;
+                }
+            }
+            None => break,
+        }
+    }
+    true
+}
+
+fn at_least_two(a: bool, b: bool, c: bool) -> bool {
+    if a {
+        b || c
+    } else {
+        b && c
+    }
+}
+
 impl UsageTracker {
     fn save(&mut self) -> Result<(), RitaCommonError> {
+        let minimum_number_transactions = 75;
```

Comment:
this save function should be called in the save_to_disk_loop

Reply:
This is called in save_to_disk_loop now 👍🏻 

make the number of transactions and bytes that need to be downloaded/uploaded into arguments, this way we can set different thresholds up in save_to_disk_loop